src/cardboard-button.js

```diff
@@ -4,11 +4,79 @@ import videojs from 'video.js';
 const Button = videojs.getComponent('Button');
 
 class CardboardButton extends Button {
+  constructor(player, options) {
+    super(player, options);
+
+    window.addEventListener('vrdisplayactivate', () => this.handleActivate_());
+    window.addEventListener('vrdisplaydeactivate', () => this.handleDeactivate_());
+
+    // vrdisplaypresentchange does not fire activate or deactivate
+    // and happens when hitting the back button during cardboard mode
+    // so we need to make sure we stay in the correct state by
+    // listening to it and checking if we are presenting it or not
+    window.addEventListener('vrdisplaypresentchange', () => {
+      if (!player.vr.vrDisplay.isPresenting && this.active_) {
+        this.handleDeactivate_();
+      }
+      if (player.vr.vrDisplay.isPresenting && !this.active_) {
+        this.handleActivate_();
+      }
+    });
+
+    // we cannot show the cardboard button in fullscreen on
+    // android as it breaks the controls, and makes it impossible
+    // to exit cardboard mode
+    if (videojs.browser.IS_ANDROID) {
+      player.on('fullscreenchange', () => {
+        if (player.isFullscreen()) {
+          this.hide();
+        } else {
+          this.show();
+        }
+      });
+    }
+  }
+
   buildCSSClass() {
     return `vjs-button-vr ${super.buildCSSClass()}`;
   }
-  handleClick() {
-    window.dispatchEvent(new window.Event('vrdisplayactivate'));
+
+  handleActivate_() {
```

Comment:
One thing to look into is whether `player.enterFullWindow` and `player.exitFullWindow` work for this usecase.